fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/FullUnorderedScanExpression.java

```diff
@@ -134,7 +143,15 @@ public String toString() {
     @Nonnull
     @Override
     public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candidateExpression, @Nonnull final AliasMap aliasMap, @Nonnull final IdentityBiMap<Quantifier, PartialMatch> partialMatchMap) {
-        return exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
+        if (getClass() != candidateExpression.getClass()) {
+            return ImmutableList.of();
+        }
+        // if 1) query contains candidate's indexes; AND 2) the query can be exactly subsumed by the candidate, the query can be subsumed by the candidate
+        if (getIndexSet().isEmpty() || getIndexSet().containsAll(((FullUnorderedScanExpression)candidateExpression).getIndexSet())) {
```

Comment:
I understand what this is doing but this goes back to my comment earlier. The fact we assign special meaning to the empty set here is not great.